### PR TITLE
Major cleanup to avoid deprecated code/warnings

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

File: sstmac/hardware/nic/nic.cc [hunk 1]
```diff
@@ -106,10 +106,10 @@ NIC::NIC(uint32_t id, SST::Params& params, Node* parent) :
   top_ = Topology::staticTopology(params);
 
   std::string subname = sprockit::sprintf("NIC.%d", my_addr_);
-  auto* spy = parent->registerMultiStatistic<int,uint64_t>(params, "spy_bytes", subname);
+  auto* spy = registerMultiStatistic<int,uint64_t>(params, "spy_bytes", subname);
   spy_bytes_ = dynamic_cast<StatSpyplot<int,uint64_t>*>(spy);
```
Comment (calewis) [inline sstmac/hardware/nic/nic.cc]: If you know the actual type of spy, can you not just static_cast here?

File: sstmac/hardware/snappr/snappr_nic.cc [hunk 2]
```diff
@@ -80,7 +80,7 @@ SnapprNIC::SnapprNIC(uint32_t id, SST::Params& params, Node* parent) :
   SST::Params inj_params = params.find_scoped_params("injection");
 
   //configure for a single port for now
-  outports_.reserve(1);
+  outports_.resize(1);
```
Comment (calewis) [inline sstmac/hardware/snappr/snappr_nic.cc]: Resize outports_ to num_ports, that way when you change the number of ports in the future the loop isn't broken.

File: sstmac/hardware/snappr/snappr_switch.cc [hunk 3]
```diff
@@ -176,17 +176,17 @@ SnapprSwitch::SnapprSwitch(uint32_t id, SST::Params& params) :
   TimeDelta byte_delay(1.0/link_bw_);
   std::string sw_arbtype = params.find<std::string>("arbitrator", "fifo");
   std::string link_arbtype = link_params.find<std::string>("arbitrator", sw_arbtype);
-  outports_.reserve(top_->maxNumPorts());
+  outports_.resize(top_->maxNumPorts());
```
Comment (calewis) [inline sstmac/hardware/snappr/snappr_switch.cc]: Optimizer _probably_ fixes this but you should do something like:
```cpp
auto maxPorts = top_->maxNumPorts();
outports_.resize(maxPorts);
inports_.resize(maxPorts);
for(int i = 0; i < maxPorts; ++i){
```
Then you can be sure you don't trigger 3 loads for a single value. Up to you though just a suggestion.

File: sstmac/common/event_scheduler.h [hunk 4]
```diff
@@ -106,7 +105,7 @@ namespace Event {
 using HandlerBase = sstmac::EventHandler;
 }
 using sstmac::Component;
-using BaseComponent = sstmac::EventScheduler;
+//using BaseComponent = sstmac::EventScheduler;
```
[the remark this answers is not in the document]
Comment (jjwilke) [reply sstmac/common/event_scheduler.h]: Done

File: sstmac/hardware/nic/nic.cc (hunk 1 above)
Comment (jjwilke) [reply sstmac/hardware/nic/nic.cc]: Added comments

File: sstmac/hardware/snappr/snappr_nic.cc (hunk 2 above)
Comment (jjwilke) [reply sstmac/hardware/snappr/snappr_nic.cc]: Changed

File: sstmac/hardware/snappr/snappr_switch.cc (hunk 3 above)
Comment (jjwilke) [reply sstmac/hardware/snappr/snappr_switch.cc]: Sure, worry about this later. Won't fix for now.

**Pull request merged**